### PR TITLE
Fix read temperature from minibuffer

### PR DESCRIPTION
**Pull request by tshu-w**
The returned result now is a string instead of a number, and the code is from `set-variable`.

### CODE REVIEW

**Comment by karthink** [comment]
By setting the `read` argument to `t`, the returned result is now a _number_, not a string.  This is useful in that we can get rid of `gptel--numberize`, but then reading in`gptel-max-tokens` should also be fixed.

**Comment by tshu-w** [comment]
Hi, I have updated my PR with `gptel--numberize` removement.

File: gptel-transient.el [hunk 1]
```diff
@@ -268,20 +272,17 @@ will get progressively longer!"
   :variable 'gptel-model
   :key "m"
   :choices '("gpt-3.5-turbo" "gpt-3.5-turbo-16k" "gpt-4" "gpt-4-32k")
-  :reader (lambda (prompt &rest _)
-            (completing-read
-             prompt
-             '("gpt-3.5-turbo" "gpt-3.5-turbo-16k" "gpt-4" "gpt-4-32k"))))
+  :reader 'gptel--transient-read-variable)
```
Comment (tshu-w) [inline gptel-transient.el:275]: I accident change this, will remove tomorrow

File: gptel.el [hunk 2]
```diff
@@ -253,18 +253,14 @@ By default, `gptel-host' is used as HOST and \"apikey\" as USER."
...
-;; FIXME Should we utf-8 encode the api-key here? 
+;; FIXME Should we utf-8 encode the api-key here?
 (defun gptel--api-key ()
   "Get api key from `gptel-api-key'."
   (pcase gptel-api-key
     ((pred stringp) gptel-api-key)
     ((pred functionp) (funcall gptel-api-key))
     (_ (error "`gptel-api-key' is not set"))))
 
-(defsubst gptel--numberize (val)
```
Comment (karthink) [inline gptel.el:264]: Please add `gptel--numberize` back in, as I need this elsewhere in the code.  The rest of the changes look good!

**Review by karthink — CHANGES_REQUESTED** [review]

File: gptel.el (hunk 2 above)
Comment (tshu-w) [reply gptel.el:264]: Hello, it seems that I have already removed all the places where `gptel--numberize` was used. Is this function no longer being used in the entire project?

Comment (karthink) [reply gptel.el:264]: Yes, it is no longer being used. I will need to use it again in the future though, so it will be better to just leave it in.

Comment (tshu-w) [reply gptel.el]: OK, I have added this function back ;)

**Comment by karthink** [comment]
Cheers @tshu-w

**Pull request merged**